### ISSUE TITLE
Security : remove access token in request query param

### ISSUE THREAD

**Issue by laurent-dol** [matrix-org/pantalaimon#116]
https://github.com/matrix-org/pantalaimon/blob/cd36ca68d5edb2d33f5a015ab04d8be4bf037912/pantalaimon/daemon.py#L182

The access_token is thrown to server Synapse in query param when a call to Api.whoami is done.
For security reasons => must be thrown in Authorization header instead.

**Comment by laurent-dol** [comment]
Replacement code:

                    method = "GET"
                    path = Api._build_path(path=["account", "whoami"])
                    headers = [("Authorization", f"Bearer {access_token}")]

                    resp = await session.request(
                        method,
                        self.homeserver_url + path,
                        proxy=self.proxy,
                        ssl=self.ssl,
                        headers=headers,
                    )

**Comment by chookity-pokk** [comment]
Hey @laurent-treeb, sorry this wasn't addressed. Can you explain the issue here though? I'm a bit ignorant on the topic here so more info would be helpful.

**Comment by laurent-treeb** [comment]
Hi, 3 years later....i have to remember.

It's a security failure, the bearer token is set in query parameters of the whoami request.
The bearer should be set in headers. I suggested a fix in my second comment.

but since 3 years maybe the code has already changed.

**Comment by chookity-pokk** [comment]
Right, I guess my question is why is it a security issue and how is it fixed in the code suggestion. For what it's worth, the code is still the same so this would work. I'm just trying to understand fundamentally what the issue is and how it is solved with the provided code since this is out of my wheelhouse.
